rules/prefer-module.js

```diff
@@ -0,0 +1,318 @@
+		yield removeSpacesAfter(equalToken, sourceCode, fixer);
+
+		yield * removeParentheses([node.parent, node], fixer, sourceCode);
+	};
+}
+
+function fixNamedExport(node, sourceCode) {
+	return function * (fixer) {
+		const assignmentExpression = node.parent.parent;
+		const exported = node.parent.property.name;
+		const local = assignmentExpression.right.name;
+		yield fixer.replaceText(assignmentExpression, `export {${local} as ${exported}}`);
+
+		yield * removeParentheses(assignmentExpression, fixer, sourceCode);
+	};
+}
+
+function fixExports(node, sourceCode) {
+	// `exports = bar`
+	if (isTopLevelAssignment(node)) {
+		return fixDefaultExport(node, sourceCode);
+	}
+
+	// `exports.foo = bar`
+	if (isNamedExport(node)) {
+		return fixNamedExport(node, sourceCode);
+	}
+}
+
+function fixModuleExports(node, sourceCode) {
+	if (isModuleExports(node)) {
+		return fixExports(node.parent, sourceCode);
+	}
+}
+
+function create(context) {
+	const sourceCode = context.getSourceCode();
+
+	return {
+		'ExpressionStatement[directive="use strict"]'(node) {
+			context.report({
+				node,
+				messageId: ERROR_USE_STRICT_DIRECTIVE,
+				* fix(fixer) {
+					yield fixer.remove(node);
+					yield removeSpacesAfter(node, sourceCode, fixer);
+				}
+			});
+		},
+		'ReturnStatement:not(:function ReturnStatement)'(node) {
+			context.report({
+				node: sourceCode.getFirstToken(node),
+				messageId: ERROR_GLOBAL_RETURN
+			});
+		},
+		[identifierSelector](node) {
+			if (isShadowed(context.getScope(), node)) {
+				return;
+			}
+
+			const {name} = node;
+
+			const problem = {
+				node,
+				messageId: ERROR_IDENTIFIER,
+				data: {name}
+			};
+
+			switch (name) {
+				case '__filename':
+				case '__dirname': {
+					const messageId = node.name === '__dirname' ? SUGGESTION_DIRNAME : SUGGESTION_FILENAME;
+					const replacement = node.name === '__dirname' ?
+						'path.dirname(url.fileURLToPath(import.meta.url))' :
+						'url.fileURLToPath(import.meta.url)';
+					problem.suggest = [{
+						messageId,
+						fix: fixer => renameIdentifier(node, replacement, fixer, sourceCode)
+					}];
+					break;
+				}
+
+				case 'require': {
+					const fix = fixRequireCall(node, sourceCode);
+					if (fix) {
+						problem.suggest = [{
+							messageId: SUGGESTION_IMPORT,
+							fix
+						}];
+					}
+
+					break;
+				}
+
+				case 'exports': {
+					const fix = fixExports(node, sourceCode);
+					if (fix) {
+						problem.suggest = [{
+							messageId: SUGGESTION_EXPORT,
+							fix
+						}];
+					}
+
+					break;
+				}
+
+				case 'module': {
+					const fix = fixModuleExports(node, sourceCode);
+					if (fix) {
+						problem.suggest = [{
+							messageId: SUGGESTION_EXPORT,
+							fix
+						}];
+					}
+
+					break;
+				}
+
+				default:
+			}
+
+			context.report(problem);
+		}
+	};
+}
+
+module.exports = {
+	create,
+	meta: {
+		type: 'suggestion',
+		docs: {
+			description: 'Prefer ES modules over CommonJS.',
```

Comment:
```suggestion
			description: 'Prefer JavaScript modules over CommonJS.',
```

Should we go with this naming?